pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -348,8 +348,12 @@ func (c *Cluster) startOSDDaemon(nodeName string, config *provisionConfig, confi
 				}
 				continue
 			}
-			logger.Infof("deployment for osd %d already exists", osd.ID)
+			logger.Infof("deployment for osd %d already exists. updating if needed", osd.ID)
+			if err = k8sutil.UpdateDeploymentAndWait(c.context, dp, c.Namespace); err != nil {
+				config.addError(fmt.Sprintf("failed to update osd deployment %d. %+v", osd.ID, err))
```

Comment:
we call `config.addError`, but we don't call `continue`, so this increments the succeeded count? is that the right thing to do?